### ISSUE TITLE
lint: add index-only range loops checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#72]
If range loop is iterating over `container` which has pointer-type elements (that is, any `*T` or reference-like type as `map` or `chan`) using only `key` variable (no `value`) and `container[key]` occurs more than once inside loop body, suggest using `for key, v := range container` form.

This code triggers a warning:
```go
func closeFiles(files []*os.File) {
	for i := range files {
		if files[i] != nil {
			files[i].Close()
		}
	}
}
```
This is suggested form:
```go
func closeFilesSuggester(files []*os.File) {
	for _, f := range files {
		if f != nil {
			f.Close()
		}
	}
}
```

**Comment by NexoMichael** [comment]
I have faced several times with this non-understanding from developers when "for range" copies full objects and when not.

I can add linter for that if nobody else working on that.

**Comment by Quasilyte** [comment]
We have `rangeExprCopy` check that detects such copies, if I get you right.

**Comment by NexoMichael** [comment]
> We have rangeExprCopy check that detects such copies, if I get you right.

Yes, I see it, I mean that I can contribute to the index-only range loops checker if nobody works on it already.

**Comment by Quasilyte** [comment]
Nobody, AFAIK. 
Could be non-trivial to implement properly though.

**Comment by cristaloleg** [comment]
There is no `in progress` label and no assignee. So it's untouched ¯\_(ツ)_/¯

**Comment by NexoMichael** [comment]
@cristaloleg ok, then I will look at it. Don't know thou how to make it assigned on me...

**Comment by cristaloleg** [comment]
@NexoMichael feel free to take it 😉
